### DISCUSSION TITLE
How should I return the pointer from rust for create Uint8List in flutter?

### DISCUSSION THREAD

**Discussion by brandon3343** [fzyzcjy/flutter_rust_bridge#1866] [Q&A] [answered]
**Problem:** 
I want to return a pointer to flutter from rust, but it always generates opaque types and I can't get the  the pointer. 
**Scenario:** 
1. There is a cache at rust that holds various images (**Vec\<u8>**), it may be accessed multiple times, so  only gets the image reference (**&Vec\<u8>**). 
2. i want  return &Vec<u8> pointer(***const u8**) to flutter for create Uint8List, but it always generates opaque type in flutter.

```rust
pub async fn get_cache(&self, key: String) -> *const u8 {
    self.cache.get(&key).unwrap().as_ptr()
}
```

```dart
final vecPointerFromRust = getCache("key") ; // Can't get pointer address Pointer<Uint8>, it's opaque type.
pointer = Pointer.fromAddress(vecPointerFromRust); 
Uint8List image = pointer.asTypedList(length); 
```

**Comment by fzyzcjy** [comment, marked as answer]
If you are confident about this (operating on raw pointers can be quite dangerous!), feel free to convert `*const u8` into an `i64` (`int`), since it is nothing but an address.

**Reply by brandon3343** [reply]
```rust
pub struct VecU8Ref {
    pub addr: usize,
    pub length: usize,
}
```
it's works, love you and love you and love you!

**Reply by fzyzcjy** [reply]
Happy to see it works!
